### ISSUE TITLE
[no-var-requires] Allow a regex for requires in no-var-requires and no-require-imports

### ISSUE THREAD

**Issue by intrepidOlivia** [typescript-eslint/typescript-eslint#3023]
Note: This issue is adapted from an [issue filed in the old TSLint repo](https://github.com/palantir/tslint/issues/3392) back in 2017. It is still applicable to my use case in 2021, so I am attempting to "re-open" it.

- [ X ] I have tried restarting my IDE and the issue persists.
- [ X ] I have updated to the latest version of the packages.
- [ X ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-var-requires": "error"
  }
}
```

```TS
const styles = require("./RootStyles.css");

export const Text = () => <span className={styles.text}>Hello world!</span>;
```

**Expected Result**

It's hard to configure CSS modules so that the VS Code TypeScript server understands them. Since they don't give great type definitions out of the box, sometimes the old style var requires are the best option.

Instead of disabling the rules globally, I'd love to have something like what the old `no-import-side-effect` had:

```
"no-var-requires": ["error", {
    "ignore-module": "(\\.css)$"
}]
```

**Actual Result**

```
Require statement not part of import statement. (@typescript-eslint/no-var-requires)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.29.0` |
| `@typescript-eslint/parser`        | `2.29.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |

**Comment by bradzacher** [comment]
You shouldn't be using `const` requires.

Use `import variable = require('..')` instead.
